### ISSUE TITLE
Enhancement: [no-unused-vars] False-positive when referencing imported value in a TSDoc

### ISSUE THREAD

**Issue by anton-johansson** [typescript-eslint/typescript-eslint#5017]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unused-vars

### Description

I propose that the rule should not give a warning if the imported type is being referenced in a TSDoc. It's very useful, specially in combination with `@deprecated Use {@link myOtherType} instead.`.

### Fail

```typescript
import myNotDeprecatedFunction from "./other-file.ts";

/**
 * @deprecated Use myNotDeprecatedFunction instead.
 */
export const myDeprecatedFunction = () => {
    console.log("Hello world");
};
```

### Pass

```typescript
import myNotDeprecatedFunction from "./other-file.ts";

/**
 * @deprecated Use {@link myNotDeprecatedFunction} instead.
 */
export const myDeprecatedFunction = () => {
    console.log("Hello world");
};
```

### Additional Info

TypeScript has a similar check, and I've created a similar issue for that too:
https://github.com/microsoft/TypeScript/issues/49173

**Comment by Josh-Cena** [comment]
I've already created an FR here: https://github.com/gajus/eslint-plugin-jsdoc/issues/858

I think this belongs better to JSDoc because AFAIK TS-ESLint never handles JSDoc.

**Comment by anton-johansson** [comment]
> I've already created an FR here: [gajus/eslint-plugin-jsdoc#858](https://github.com/gajus/eslint-plugin-jsdoc/issues/858)
> 
> I think this belongs better to JSDoc because AFAIK TS-ESLint never handles JSDoc.

Cool, thanks for that link! I'm not sure where it best belongs, I'm assuming the rule in this project needs to be tweaked as well?

**Comment by Josh-Cena** [comment]
If that one got implemented as I proposed—no. `markVariableAsUsed` will propagate to every other plugin. This is also how `eslint-plugin-vue` is able to mark variables used in the template as used, IIRC.

**Comment by anton-johansson** [comment]
> If that one got implemented as I proposed—no. `markVariableAsUsed` will propagate to every other plugin. This is also how `eslint-plugin-vue` is able to mark variables used in the template as used, IIRC.

Aha, I understand. That's great, I think we can close this issue then.

**Comment by bradzacher** [comment]
Yup - we do not support code within JSDoc comments because it is not real code.

Josh is correct. Any other plugin can mark a variable as used if it desires and our rule will respect it.
